// unenv/runtime/node/timers/index.mjs
import { notImplemented } from "../../_internal/utils.mjs";
import noop from "../../mock/noop.mjs";
import promises from "./promises/index.mjs";
import { setTimeoutFallback } from "./internal/set-timeout.mjs";
import {
  setImmediateFallback,
  clearImmediateFallback
} from "./internal/set-immediate.mjs";
import { setIntervalFallback } from "./internal/set-interval.mjs";
export * as promises from "./promises/index.mjs";
export const clearImmediate = globalThis.clearImmediate?.bind(globalThis) || clearImmediateFallback;
export const clearInterval = globalThis.clearInterval?.bind(globalThis) || noop;
export const clearTimeout = globalThis.clearTimeout?.bind(globalThis) || noop;
export const setImmediate = globalThis.setImmediate?.bind(globalThis) || setImmediateFallback;
export const setTimeout = globalThis.setTimeout?.bind(globalThis) || setTimeoutFallback;
export const setInterval = globalThis.setInterval?.bind(globalThis) || setIntervalFallback;
export const active = function active2(timeout) {
  timeout?.refresh?.();
};
export const _unrefActive = active;
export const enroll = notImplemented("timers.enroll");
export const unenroll = notImplemented("timers.unenroll");
export default {
  // @ts-expect-error deprecated
  _unrefActive,
  active,
  clearImmediate,
  clearInterval,
  clearTimeout,
  enroll,
  promises,
  setImmediate,
  setInterval,
  setTimeout,
  unenroll
};
